### ISSUE TITLE
[BUG] AmplitudeEmbedding for GPU

### ISSUE THREAD

**Issue by toshi-koike-akino** [PennyLaneAI/pennylane#2419]
### Expected behavior

We want to use qml.AmplitudeEmbedding on GPU. It works if inputs do not need padding, but padding is not compatible with GPU yet.

### Actual behavior

Error happens.

### Source code

```shell
import pennylane as qml
import torch

n_wires = 2
dev = qml.device('default.qubit', wires=n_wires)
@qml.qnode(dev)
def circuit_cuda(inputs):
    qml.AmplitudeEmbedding(inputs, wires=range(n_wires), pad_with=0, normalize=True)
    return qml.expval(qml.PauliZ(0))

inputs = torch.rand(n_wires) # embedding shorter than 2 ** n_wires
print(circuit_cuda(inputs)) # it works on cpu

inputs = inputs.to(torch.device('cuda')) # move to GPU
print(circuit_cuda(inputs)) # it fails
```

### Tracebacks

```shell
Traceback (most recent call last):
  File "tes.py", line 15, in <module>
    print(circuit_cuda(inputs))
  File "/lib/python3.9/site-packages/pennylane/qnode.py", line 568, in __call__
    self.construct(args, kwargs)
  File "/lib/python3.9/site-packages/pennylane/qnode.py", line 488, in construct
    self._qfunc_output = self.func(*args, **kwargs)
  File "tes.py", line 8, in circuit_cuda
    qml.AmplitudeEmbedding(inputs, wires=range(n_wires), pad_with=0, normalize=True)
  File "/lib/python3.9/site-packages/pennylane/templates/embeddings/amplitude.py", line 129, in __init__
    features = self._preprocess(features, wires, pad_with, normalize)
  File "/lib/python3.9/site-packages/pennylane/templates/embeddings/amplitude.py", line 207, in _preprocess
    feature_set = qml.math.concatenate([feature_set, padding], axis=0)
  File "/lib/python3.9/site-packages/pennylane/math/multi_dispatch.py", line 178, in wrapper
    return fn(*args, **kwargs)
  File "/lib/python3.9/site-packages/pennylane/math/multi_dispatch.py", line 259, in concatenate
    return np.concatenate(values, axis=axis, like=like)
  File "/lib/python3.9/site-packages/autoray/autoray.py", line 84, in do
    return get_lib_fn(backend, fn)(*args, **kwargs)
  File "/lib/python3.9/site-packages/autoray/autoray.py", line 409, in translated_function
    return fn(**new_kwargs)
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument tensors in method wrapper___cat)
```

### System information

```shell
Name: PennyLane
Version: 0.22.2
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /lib/python3.9/site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, retworkx, scipy, semantic-version, toml
Required-by: PennyLane-Lightning
Platform info:           Linux-5.13.0-30-generic-x86_64-with-glibc2.31
Python version:          3.9.12
Numpy version:           1.22.3
Scipy version:           1.8.0
Installed devices:
- default.gaussian (PennyLane-0.22.2)
- default.mixed (PennyLane-0.22.2)
- default.qubit (PennyLane-0.22.2)
- default.qubit.autograd (PennyLane-0.22.2)
- default.qubit.jax (PennyLane-0.22.2)
- default.qubit.tf (PennyLane-0.22.2)
- default.qubit.torch (PennyLane-0.22.2)
- lightning.qubit (PennyLane-Lightning-0.22.1)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by CatalinaAlbornoz** [comment]
Hi @toshi-koike-akino , thank you for opening this bug. What Torch version are you using? Also, does your problem get fixed if you use the [pennylane-torch interface](https://pennylane.readthedocs.io/en/stable/introduction/interfaces/torch.html)? 
You can use it by adding the keyword in the definition of your qnode: `@qml.qnode(dev, interface='torch')`

**Comment by toshi-koike-akino** [comment]
> print(torch.__version__)
1.10.2

I confirmed that interface='torch' does not resolve the issue.

**Comment by mlxd** [comment]
Thanks for reporting this @toshi-koike-akino . This failure is coming from a mixture of numpy data and Torch tensors in the concatenation dispatcher method. I can confirm that explicitly copying data back to the CPU within this method at this point: https://github.com/PennyLaneAI/pennylane/blob/3af8ae3e3c34a03ee502801a55deedb418b2b1ee/pennylane/math/multi_dispatch.py#L251 solves the problem. We will look into generalizing this so that we can keep the GPU performance whilst still allowing the same functionality with `AmplitudeEmbedding`.

**Comment by mlxd** [comment]
Hi @toshi-koike-akino thanks again for reporting this. The issue will be fixed in the upcoming 0.23 PennyLane release, and will be addressed by the PR https://github.com/PennyLaneAI/pennylane/pull/2431/
